### PR TITLE
[cairo]Add feature xrandr

### LINKED ISSUE #22338: [WSL] opencv fails to install on ubuntu - fails to compile cairo
steps to reproduce:

`vcpkg install opencv`

```
Error: Building package cairo:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+cairo
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[cairo]+Build+error
including:
  package: cairo[core,fontconfig,freetype,gobject,x11]:x64-linux -> 1.17.4#3
    vcpkg-tool version: 202
```

### PR DESCRIPTION
**Pull request by Adela0814**
Fixes: #22338
Add feature `xrandr` and add message to prompt the necessary dependencies `xrandr` and `libxext-dev`

### CODE REVIEW

**Review by JackBoosY — DISMISSED** [review]

**Review by dg0yt — CHANGES_REQUESTED** [review]
Please fix the PR title. It will be part of the final commit message. It is "cairo"

**Comment by Adela0814** [comment]
> Please fix the PR title. It will be part of the final commit message. It is "cairo"
Fixed, thanks for reminding

**Review by dg0yt — DISMISSED** [review]

File: ports/cairo/portfile.cmake [hunk 1]
```diff
@@ -27,14 +27,21 @@ if ("x11" IN_LIST FEATURES)
...
     list(APPEND OPTIONS -Dxlib=disabled)
 endif()
 list(APPEND OPTIONS -Dxcb=disabled)
 #list(APPEND OPTIONS -Dxlib-xcb=disabled) don't forget this option with the next update!
 
+if ("xrandr" IN_LIST FEATURES)
+    if (VCPKG_TARGET_IS_WINDOWS)
+        message(FATAL_ERROR "Feature xrandr only support UNIX.")
+    endif()
+    message(WARNING "You will need to install Xrandr dependencies to use feature xrandr:\napt install libxrandr-dev\n")
```
Comment (strega-nil-ms) [inline ports/cairo/portfile.cmake:42]: This doesn't seem to actually enable xrandr support?

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
Is there something I'm missing?

File: ports/cairo/portfile.cmake (hunk 1 above)
Comment (Adela0814) [reply ports/cairo/portfile.cmake:42]: yes, this just a message, user need to use 'apt' to install xrandr

Comment (JackBoosY) [reply ports/cairo/portfile.cmake:42]: Ping @strega-nil-ms .

**Comment by BillyONeal** [comment]
Agree with @vicroms : the port needs to be fixed to inject the selection of the feature into the build system rather than just hoping the right thing gets found from the system

**Pull request closed without merging**